src/Microsoft.AspNetCore.Sockets/SocketBuilderExtensions.cs

```diff
@@ -12,12 +9,9 @@ public static class SocketBuilderExtensions
     {
         public static ISocketBuilder UseEndPoint<TEndPoint>(this ISocketBuilder socketBuilder) where TEndPoint : EndPoint
         {
+            var endpoint = socketBuilder.ApplicationServices.GetRequiredService<TEndPoint>();
```

Comment:
Was there any specific reason you changed it? Asking because you extended scope of endpoint which is technically not needed,